flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -832,14 +833,42 @@ private StateBackend loadStateBackend(ReadableConfig configuration, ClassLoader
 	 * @param to
 	 * 		The number to stop at (inclusive)
 	 * @return A data stream, containing all number in the [from, to] interval
+	 * @deprecated Use {@link #fromSequence(long, long)} instead to create a new data stream
+	 * that contains {@link org.apache.flink.api.connector.source.lib.NumberSequenceSource}.
 	 */
+	@Deprecated
 	public DataStreamSource<Long> generateSequence(long from, long to) {
 		if (from > to) {
 			throw new IllegalArgumentException("Start of sequence must not be greater than the end");
 		}
 		return addSource(new StatefulSequenceSource(from, to), "Sequence Source");
 	}
 
+	/**
+	 * Creates a new data stream that contains {@link org.apache.flink.api.connector.source.lib.NumberSequenceSource}
```

Comment:
I don't think we need to mention that implementation detail here.

@StephanEwen Sorry if you requested that this be mentioned here, then we can leave it in. 😅

Reply:
@StephanEwen Does this need to mention that implementation detail? I thought this need be mentioned here.